### PR TITLE
fix: model plugin support component data

### PR DESCRIPTION
**Pull request by liuxi150**
## Summary by CodeRabbit

- **New Features**
	- Expanded type declarations for various model components, enhancing modeling capabilities.
	- Introduced changes to handle shader compilation and material initialization more efficiently.
	- Updated the handling of emissive colors in shaders to support more accurate color calculations.

- **Refactor**
	- Updated model component classes to use new data types and naming conventions for clarity and consistency.
	- Modified material and mesh runtime logic for improved performance and flexibility.
	- Adjusted shader library code to accommodate changes in emissive color handling.

- **Chores**
	- Added a new URL reference for scene JSON file to align with updated data sources.
	- Implemented a check for WebGL2 support to ensure compatibility with shader creation requirements.

### CODE REVIEW

File: plugin-packages/model/src/runtime/material.ts [hunk 1]
```diff
@@ -492,7 +502,7 @@ export class PMaterialPBR extends PMaterialBase {
   /**
    * 自发光颜色值，默认是黑色 Vector3(0, 0, 0)
```
Comment (yiiqii) [inline plugin-packages/model/src/runtime/material.ts:503]: 注释也更新一下吧

**Review by yiiqii — CHANGES_REQUESTED** [review]

File: plugin-packages/model/src/utility/plugin-helper.ts [hunk 2]
```diff
@@ -30,6 +30,7 @@ import {
   DestroyOptions,
   loadImage,
   PLAYER_OPTIONS_ENV_EDITOR,
+  GLSLVersion,
```
[the remark this answers is not in the document]
Comment (liuxi150) [reply plugin-packages/model/src/utility/plugin-helper.ts:33]: import不需要合并

File: plugin-packages/model/src/runtime/mesh.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
-import type { Texture, Geometry, Engine, math, VFXItemContent, VFXItem, Renderer } from '@galacean/effects';
-import { spec, Mesh, DestroyOptions, Material } from '@galacean/effects';
+import type { Texture, Engine, math, VFXItemContent, VFXItem, Renderer } from '@galacean/effects';
+import { Geometry, spec, Mesh, DestroyOptions, Material, GLSLVersion, Shader } from '@galacean/effects';
```
[the remark this answers is not in the document]
Comment (liuxi150) [reply plugin-packages/model/src/runtime/mesh.ts:2]: Geometry和GLSLVersion有被用到

File: plugin-packages/model/src/runtime/mesh.ts [hunk 4]
```diff
@@ -511,19 +511,41 @@
...
+        {
+          shader: {
+            vertex: this.material.vertexShaderCode,
+            fragment: this.material.fragmentShaderCode,
+            shared: globalState.shaderShared,
+            glslVersion: isWebGL2 ? GLSLVersion.GLSL3 : GLSLVersion.GLSL1,
+          },
+          uniformSemantics: newSemantics,
+        }
+      );
```
[the remark this answers is not in the document]
Comment (liuxi150) [reply plugin-packages/model/src/runtime/mesh.ts:547]: 临时兼容代码，先这么处理，后面再优化

File: plugin-packages/model/src/runtime/material.ts (hunk 1 above)
Comment (liuxi150) [reply plugin-packages/model/src/runtime/material.ts]: 已经更新

**Review by yiiqii — APPROVED** [review]

**Pull request merged**